src/pages/managerDashboard.js

```diff
@@ -1,15 +1,17 @@
-import React from 'react';
-import Link from 'next/link';
-import NavBar from '@/components/Header/NavBar';
-import CustomizableAvatar from '@/components/common/customavatar/CustomizableAvatar';
-import CustomizableButton from '@/components/common/CustomizableButton';
-import FooterBar from '@/components/common/FooterBar';
-import styles from '../styles/singleListing.module.css';
+import React, { useEffect } from "react";
+import { useRouter } from "next/router";
+import useAuthContext from "@/context/useAuthContext";
+const ManagerDashboard = () => {
+  const { state } = useAuthContext();
+  const  router = useRouter();
 
-const managerDashboard = () => {
-  return (
-    <div>managerDashboard</div>
-  )
-}
+  useEffect(() => {
+    if (state?.user ===undefined || state?.user?.uid === "") {
+    router.push("/");
```

Comment:
![auth-guard-jump](https://github.com/AritDeveloperCircle/findyourpeople/assets/32520970/fe082cfb-8591-4b9a-920c-146ce9844dab)

So @ibimina it appears that your `router.push("/")` implementation here is causing a noticeable "jump" in the user experience (view the GIF above).

I lack the NextJS knowledge so forgive me, but it it possible to render the managerDashboard child components conditionally instead? And then if the logged-in condition is unmet, render an error message to the user, instead of the redirect to index?

@annonymousauthority Please weigh in if you can. 🙏🏾

Reply:
She's trying to create a route guard.

@ibimina would you consider making the components group that requires Auth to show loading components/like @msarit said, a would you like to signing screen?

To make it feel natural.

A route guard might be brute for this case.